### PR TITLE
Add `ambulance` icon

### PR DESCRIPTION
**Pull request by jordan808**
## What is the purpose of this pull request?

- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description
Adds ambulance icon as requested by https://github.com/lucide-icons/lucide/issues/119. Closes feather icons issue https://github.com/feathericons/feather/issues/1177.

### Icon use case 

For navigation applications directing users to the nearest emergency services, or for listing an emergency contact.

### Alternative icon designs 
ambulance-alternative

### Concept 

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [ ] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [x] I've based them on the following Lucide icons: Bus
- [ ] I've based them on the following design: 

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to two points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

File: icons/ambulance.svg [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  fill="none"
+  stroke="currentColor"
+  stroke-width="2"
+  stroke-linecap="round"
+  stroke-linejoin="round"
+>
+  <path
+    d="M7 20a2 2 0 1 0 0-4 2 2 0 0 0 0 4ZM17 20a2 2 0 1 0 0-4 2 2 0 0 0 0 4Z" />
+  <path
+    d="M9 18h5m0 0V7a2 2 0 0 0-2-2H4a2 2 0 0 0-2 2v10a1 1 0 0 0 1 1h2m9 0h1M8 9v4m2-2H6m13 7h2a1 1 0 0 0 1-1v-3.28a1 1 0 0 0-.68-.95l-2-.66a1 1 0 0 1-.51-.4L17.3 9.45a1 1 0 0 0-.84-.45H14" />
```
Comment (jguddas) [inline icons/ambulance.svg:15]: Had to lift up the roof and + by 1px so we don't violate the 2px gap rule.

```suggestion
  <path d="M10 10H6" />
  <path d="M14 18V6a2 2 0 0 0-2-2H4a2 2 0 0 0-2 2v11a1 1 0 0 0 1 1h2" />
  <path d="M19 18h2a1 1 0 0 0 1-1v-3.28a1 1 0 0 0-.68-.95l-2-.66a1 1 0 0 1-.51-.4L17.3 9.45a1 1 0 0 0-.84-.45H14" />
  <path d="M8 8v4" />
  <path d="M9 18h6" />
  <circle cx="17" cy="18" r="2" />
  <circle cx="7" cy="18" r="2" />
```

https://lucide-studio.vercel.app/?value=%3Csvg++xmlns=%22http://www.w3.org/2000/svg%22++width=%2224%22++height=%2224%22++viewBox=%220+0+24+24%22++fill=%22none%22++stroke=%22currentColor%22++stroke-width=%222%22++stroke-linecap=%22round%22++stroke-linejoin=%22round%22%3E++%3Cpath+d=%22M10+10H6%22+/%3E++%3Cpath+d=%22M14+18V6a2+2+0+0+0-2-2H4a2+2+0+0+0-2+2v11a1+1+0+0+0+1+1h2%22+/%3E++%3Cpath+d=%22M19+18h2a1+1+0+0+0+1-1v-3.28a1+1+0+0+0-.68-.95l-2-.66a1+1+0+0+1-.51-.4L17.3+9.45a1+1+0+0+0-.84-.45H14%22+/%3E++%3Cpath+d=%22M8+8v4%22+/%3E++%3Cpath+d=%22M9+18h6%22+/%3E++%3Ccircle+cx=%2217%22+cy=%2218%22+r=%222%22+/%3E++%3Ccircle+cx=%227%22+cy=%2218%22+r=%222%22+/%3E%3C/svg%3E

**Review by jguddas — CHANGES_REQUESTED** [review]

File: icons/ambulance.svg (hunk 1 above)
Comment (jordan808) [reply icons/ambulance.svg:15]: Makes sense. Should I commit your changes or do you want me to update from my original file?

Comment (jguddas) [reply icons/ambulance.svg:15]: You can commit it, maybe that will also fix the preview comment.

**Review by ericfennis — COMMENTED** [review]
For consistency, maybe an idea to update the `truck` design as well.
image

Maybe this is also a better shape to use in #1803

**Comment by jguddas** [comment]
I feel like it's okay for the ambulance to be different and more boxy, that's also how they are in reality.

**Comment by ericfennis** [comment]
@jguddas But I think the shape of the "truck" of @jordan808 fits way better with the lucide style, it's also vertically centered. 

100% side by side:
Frame 139

**Comment by jguddas** [comment]
> @jguddas But I think the shape of the "truck" of @jordan808 fits way better with the lucide style, it's also vertically centered. 
> 
> 100% side by side:
> Frame 139
> 

Good point.

**Comment by jordan808** [comment]
Below, I've explored some options to achieve greater consistency. The last one is my favorite, where the truck utilizes the same cab style as the ambulance icon in this PR. This choice would also be more modern, as the flat-face cab style (known as cab-overs or flat faces) has become less common in recent decades due to truck size regulations, at least in the U.S.

Here are some of my explorations aimed at enhancing consistency:

Cab heights same (taller):
ambulance-tall
truck-tall

Cab heights same (shorter):
ambulance-short
truck-short

Cab's styles same (taller):
ambulance-tall-matching
truck-tall-matching

Cab's styles same (shorter):
ambulance-short-matching
truck-short-matching

What do you all think @jguddas @ericfennis ?

**Comment by ericfennis** [comment]
@jordan808 Nice work on the designs. It's hard to see differences. But I think I like the first and the third options. It uses the space on the canvas a bit better.

**Comment by jguddas** [comment]
I like the first option the most.
image

**Comment by jordan808** [comment]
Do you all want to go with the first option then? If so, we would need to increase the cab height of the `ambulance` so that it is the same height as the `truck` if/when we update the truck icon. I'm warming up to the increased cab height anyway because it does utilize the verticle space better. 

### Proposed change to `ambulance`
ambulance-short → ambulance-tall
Increase the cab height to reflect potential adjustment to truck icon.
### Proposed change to `truck`
truck-original → truck-tall
Increase trailer size, cab height, and round corners to 2px radius ([Rule #7](https://lucide.dev/guide/design/icon-design-guide#_7-shapes-such-as-rectangles-must-have-a-border-radius-of)).
### Finalized `ambulance` and `truck` comparison
ambulance-tall truck-tall

If you all approve, I can make the updated changes to the ambulance if needed.

**Comment by ericfennis** [comment]
@jordan808 looks excellent 🤩. I'm in for the new truck icon.

File: icons/ambulance.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "../icon.schema.json",
+  "contributors": [
+    "jordan808"
```
[the remark this answers is not in the document]
Comment (jordan808) [reply icons/ambulance.json:4]: Gotcha. This was mainly based on the truck icon, so I will add everyone from that one

**Review by jordan808 — COMMENTED** [review]
Added contributors 👍

**Review by jguddas — APPROVED** [review]

**Review by ericfennis — APPROVED** [review]

**Comment by ericfennis** [comment]
@jordan808 Can you make a PR for the changes for the `truck` icon?

**Pull request merged**